python/ray/external_storage.py

```diff
@@ -128,22 +128,22 @@ def _write_multiple_objects(
         ):
             address_len = len(owner_address)
             metadata_len = len(metadata)
-            if buf is None:
-                error = f"object ref {ref.hex()} does not exist."
-                # empty data and 1 byte metadata, this object is
-                # marked as failed.
-                if metadata_len == 1:
-                    error += " This is probably since its owner has failed."
+            if buf is None and len(metadata) == 0:
+                error = f"Object {ref.hex()} does not exist."
                 raise ValueError(error)
-            buf_len = len(buf)
+            if buf is None:
+                buf_len = 0
+            else:
+                buf_len = len(buf)
             payload = (
                 address_len.to_bytes(8, byteorder="little")
                 + metadata_len.to_bytes(8, byteorder="little")
                 + buf_len.to_bytes(8, byteorder="little")
                 + owner_address
                 + metadata
-                + memoryview(buf)
             )
+            if buf is not None:
+                payload += memoryview(buf)
```

Comment:
payload is `bytes` type and it doesn't have an additional copy when we do `+=` right (unlike string)?

Reply:
Hmm not sure actually, but I'll change this to do it inline with the first allocation!